newIDE/app/src/ResourcesList/ResourceUtils.js

```diff
@@ -34,3 +43,46 @@ export const resourceHasValidPath = (
   const resourcePath = getLocalResourceFullPath(project, resourceName);
   return fs.existsSync(resourcePath);
 };
+
+export const selectLocalResourcePath = (
```

Comment:
It's a good idea to refactor this 👍
Check my comments about making it more straightforward (options should only be input options, all the output should be transferred through the promise)

Reply:
Can you move it to `LocalResourceSources.js` though?
ResourceUtils.js should be agnostic of the platform where we are running - so no mention of electron or path or dialog should be present here.
Also dont' export it (just declare it with `const`). Nobody should use directly this function. Instead, components that want to select a resource should call `onChooseResource` (which is given by mainFrame), with the kind of the resource to select.

They will be given, in the promise result, a resource that was chosen by the user. You're then free to use this resource, or extract some information like the path, and then delete it :)

Reply:
good idea! I completely missed it because you have named the function with a "_" prefix (`_onChooseResource`) , leading me to believe that it is local. Would youlike me to remove the _ prefix? :)

Reply:
The underscore is fine here because _onChooseResource is indeed private in MainFrame, i.e: it should not be accessed from the exterior of the class. But it's then passed down to other components using a prop called "onChooseResource". So it's fine for other components to use it (and it's fine for MainFrame to pass it).
Other components receive a prop called "onChooseResource" and they don't need to know where this thing is coming from : it's an implementation detail of MainFrame, which is responsible for providing the proper method :)